### DISCUSSION TITLE
Busca de profissionais por número de conselho

### DISCUSSION THREAD

**Discussion by gustavofabro** [BrasilAPI/BrasilAPI#215] [Ideas]
Olá pessoal, estive pensando em uma API que retorna dados de um profissional através da consulta do seu número de conselho, como o CRM (médicos), CREA (engenheiros), e por aí vai. 

Encontrei uma API que retorna essas informações: https://www.consultacrm.com.br/index/api. Ela possui alguns domínios específicos para cada tipo de conselho, por exemplo tem o https://www.consultaoab.com.br/index/api para consulta de registros da OAB. Mas na verdade basta trocar o tipo de conselho no parâmetro da API que vai funcionar, mesmo tu estando no domínio de outro conselho:

https://www.consultacrm.com.br/api/index.php?tipo=crea&uf=&q=Joao&chave=CHAVE&destino=json 

O problema é que ele é limitada a 100 consultas mensais no plano gratuito, mas isso poderíamos resolver com o cache das informações, imagino que dê de fazer um trabalho bem legal nesse sentido, realizando consultas diárias, e persistindo esses resultados para reutilizar em outras consultas.

Minha ideia seria criar uma API para cada conselho. Não manjo muito do Next mas me disponho a desenvolver também.

O que acham da ideia? Alguém conhece alguma outra API que retorna esse tipo de informação?

**Comment by rodbv** [comment]
Fala Gustavo! Excelente ideia, consulta de registros profissionais é muito útil.

Algumas considerações:

- É preciso ver se fazer cache local do consultacrm.com.br não infringiria os termos e condições de "uso justo"  da API deles, visto que eles são um serviço comercial. Uma outra questão secundária é que se fossemos criar um cache disso, teria que ver onde seriam armazenados esses dados, e como seria arcado esses custos de armazenamento.

- Pro caso do registro na OAB, eu acho que existe uma alternativa para essa consulta em https://cna.oab.org.br, pelo que vi eles tem um endpoint em `https://cna.oab.org.br/Home/Search` que permite chamadas POST para pesquisa e retornam os dados em uma lista JSON. O unico problema é que eles tem proteção ReCaptcha. Se esse problema do ReCaptcha for resolvido, o que poderia ser feito aqui no BrasilAPI seria criar uma fachada mais amigável para essa API? Mais ou menos como foi feito pro CEP. Pode valer a pena contatar o pessoal  da OAB e  perguntar se eles tem interesse em colaborar, liberando um endpoint sem ReCaptcha.

**Comment by RickSSousa** [comment]
Estou precisando de uma API que consulte a partir de um registro da OAB, mas ainda não consegui, nem mesmo a consultacrm.com. Vocês descobriram alguma alternativa?

**Comment by halvarezoliveira** [comment]
Fala Gustavo, tudo certo,
estou precisando também para OAB e CRMV tem alguma coisa para me ajudar? Onde teria a documentação ou se ja tem algo pronto? Procuro para PHP. Precisa criar algum token?
Abraço

**Comment by SkylineNando** [comment]
Você pode tentar por meio desse link https://api.crea-go.org.br/busca/profissionaiseempresas

**Comment by Fabriziosilveira** [comment]
Alguém sabe alguma API para consultar CREA ?

**Reply by deboramagnago** [reply]
Olá. Também estou precisando
